service/controller/v3/resource/instance/create_test.go

```diff
@@ -544,6 +544,138 @@ func Test_Resource_Instance_findActionableInstance(t *testing.T) {
 	}
 }
 
+func Test_Resource_Instance_computeForDeleteResourceStatus(t *testing.T) {
```

Comment:
Had to write some unit tests to ensure the status structures are cleaned properly and do not leave behind empty garbage structures.